audio/filter/af_scaletempo2_internals.c

```diff
@@ -41,19 +41,15 @@ static bool in_interval(int n, struct interval q)
     return n >= q.lo && n <= q.hi;
 }
 
-static float **realloc_2d(float **p, int x, int y)
+static float **alloc_sample_buffer(struct mp_scaletempo2 *p, float **ptr, size_t size)
```

Comment:
```suggestion
static void alloc_sample_buffer(struct mp_scaletempo2 *p, float ***ptr, size_t size)
```
Maybe `***` is not that nice, but at least it is explicit that we release/change this pointer.

Reply:
I found the result nice, so I added this change. I also replaced the `talloc_realloc` usages with `MP_RESIZE_ARRAY` for the same reason.